pkg/cri/resource-manager/control/cri/cri.go

```diff
@@ -86,7 +86,9 @@ func (ctl *crictl) PreCreateHook(c cache.Container) error {
 	create.Config.Envs = c.GetCRIEnvs()
 	create.Config.Mounts = c.GetCRIMounts()
 	create.Config.Devices = c.GetCRIDevices()
-	create.Config.Linux.Resources = c.GetLinuxResources()
+	if create.Config.Linux != nil {
+		create.Config.Linux.Resources = c.GetLinuxResources()
+	}
```

Comment:
Hmm.... although this should never happen if the request comes from kubelet, I still think the proper fix would be to do something like this:

```
if create.Config.Linux == nil {
    create.Config.Linux = &cri.LinuxContainerConfig{}
} 
create.Config.Linux.Resources = c.GetLinuxResources()
```

Otherwise you could in principle avoid the (CRI parts of the) policies getting enforced on your container by leaving Config.Linux deliberately nil in the create request, just like above.

Reply:
Ok, fixed.